### PR TITLE
Recursively call parent handlers when child is loggable

### PR DESCRIPTION
**Pull request by xerial**
This PR fixes a bug which discards logs even if a child logger has a finer loglevel than the parent level.

For example, when we have a parent logger (INFO) and its child logger (FINE or finer), in the current code, calling log.fine(msg) of the child logger will bypass the request to parent.log(FINE, msg). But this log message will be ignored because the parent log level is INFO. 

So instead of calling `parent.log`, we should call the parent handler's `publish` method.

### CODE REVIEW

**Comment by xerial** [comment]
@sjrd This bug is only a remaining problem I can find when using this library with https://github.com/wvlet/log. I hope a new version (0.1.1) will be released after this PR is merged.

File: src/main/scala/java/util/logging/Logger.scala [hunk 1]
```diff
@@ -107,8 +107,9 @@ class Logger(name: String, resourceBundle: String) {
     if (isLoggable(record.getLevel)) {
       if (useParentsHandlers) {
         val parent = getParent()
-        if (parent != null)
-          parent.log(record)
+        if (parent != null) {
+          parent.getHandlers.foreach(_.publish(record))
+        }
```
Comment (sjrd) [inline src/main/scala/java/util/logging/Logger.scala]: This will ignore grand-parents that would have registered handlers.

Looks like we need a `private def publish(record): Unit` in `Logger`, that would contain everything in the `{}` of `if (isLoggable(...))`, and recursively call `parent.publish(record)` rather than looking inside `parent.getHandlers`.

**Review by sjrd — CHANGES_REQUESTED** [review]
One corner case (that would need an additional test).

Also, could you squash all the commits, please? We don't do "failing test commits" in this repo; instead we want all commits in the history to pass the CI.

File: src/main/scala/java/util/logging/Logger.scala (hunk 1 above)
Comment (xerial) [reply src/main/scala/java/util/logging/Logger.scala]: That's true. I'll add more test cases and fix the logic.

**Comment by xerial** [comment]
OK. I'll squash the commits after fixing the grand parent issue.

**Comment by xerial** [comment]
@sjrd Revised the commits as suggested.

**Review by sjrd — APPROVED** [review]

**Comment by sjrd** [comment]
I'll find some time tomorrow to publish a new version. If I forget, feel free to remind me ;)
